File: examples/java/kora-java-http-server/src/main/java/io/koraframework/example/http/server/SyncController.java
package io.koraframework.example.http.server;

import io.koraframework.common.annotation.Component;
import io.koraframework.http.common.HttpMethod;
import io.koraframework.http.common.annotation.HttpRoute;
import io.koraframework.http.common.body.HttpBody;
import io.koraframework.http.server.common.response.HttpServerResponse;
import io.koraframework.http.server.common.annotation.HttpController;

/**
 * @see HttpServerResponse - Respond in sync mode with simple HttpResponse
 */
@Component
@HttpController
public final class SyncController {

    @HttpRoute(method = HttpMethod.GET, path = "/sync")
    public HttpServerResponse get() {
        return HttpServerResponse.of(200, HttpBody.plaintext("Hello world"));
    }
}

File: examples/kotlin/kora-kotlin-http-server/src/main/kotlin/io/koraframework/kotlin/example/http/server/SyncController.kt
package io.koraframework.kotlin.example.http.server

import io.koraframework.common.annotation.Component
import io.koraframework.http.common.HttpMethod
import io.koraframework.http.common.annotation.HttpRoute
import io.koraframework.http.common.body.HttpBody
import io.koraframework.http.server.common.annotation.HttpController
import io.koraframework.http.server.common.response.HttpServerResponse

@Component
@HttpController
class SyncController {
    @HttpRoute(method = HttpMethod.GET, path = "/sync")
    fun get(): HttpServerResponse = HttpServerResponse.of(200, HttpBody.plaintext("Hello world"))
}
